spec/models/product_import/inventory_reset_strategy_spec.rb

```diff
@@ -110,36 +124,44 @@
       let(:excluded_items_ids) { nil }
 
       context 'and supplier_ids is []' do
-        let(:supplier_ids) { [] }
         let(:relation) do
           instance_double(ActiveRecord::Relation, update_all: true)
         end
 
         before { allow(VariantOverride).to receive(:where) { relation } }
 
         it 'does not update any DB record' do
-          inventory_reset.reset(supplier_ids)
+          inventory_reset.reset([])
           expect(relation).not_to have_received(:update_all)
         end
       end
 
       context 'and supplier_ids is nil' do
-        let(:supplier_ids) { nil }
         let(:relation) do
           instance_double(ActiveRecord::Relation, update_all: true)
         end
 
         before { allow(VariantOverride).to receive(:where) { relation } }
 
         it 'does not update any DB record' do
-          inventory_reset.reset(supplier_ids)
+          inventory_reset.reset(nil)
           expect(relation).not_to have_received(:update_all)
         end
       end
 
       context 'and supplier_ids is set' do
+        let(:enterprise) { variant.product.supplier }
```

Comment:
same here.

Seeing this code copy pasted 3 times I wonder if you should instead create a context where you do build_stubbed() at the top and a context for the rest where you use create().